service/frontend/operator_handler_test.go

```diff
@@ -110,311 +111,378 @@ func (s *operatorHandlerSuite) TearDownTest() {
 	s.handler.Stop()
 }
 
-func (s *operatorHandlerSuite) Test_AddSearchAttributes_EmptyIndexName() {
-	handler := s.handler
+func (s *operatorHandlerSuite) Test_AddSearchAttributes() {
 	ctx := context.Background()
-
-	type test struct {
-		Name     string
-		Request  *operatorservice.AddSearchAttributesRequest
-		Expected error
-	}
-	// request validation tests
-	testCases1 := []test{
+	testCases := []struct {
+		name                      string
+		request                   *operatorservice.AddSearchAttributesRequest
+		getSearchAttributesCalled bool
+		getSearchAttributesErr    error
+		indexName                 string
+		expectedErrMsg            string
+	}{
 		{
-			Name:     "nil request",
-			Request:  nil,
-			Expected: &serviceerror.InvalidArgument{Message: "Request is nil."},
+			name:           "failed request nil",
+			request:        nil,
+			expectedErrMsg: errRequestNotSet.Error(),
 		},
 		{
-			Name:     "empty request",
-			Request:  &operatorservice.AddSearchAttributesRequest{},
-			Expected: &serviceerror.InvalidArgument{Message: "SearchAttributes are not set on request."},
+			name:           "failed search attributes not set",
```

Comment:
Could you rewrite the subtest names to be more akin to what @ast2023 did in https://github.com/temporalio/temporal/pull/5161/files ? Something like "a request without search attributes should fail" clearly specifies WHAT we're testing and HOW it should behave